plugins/git/git_commit_behind

```diff
@@ -0,0 +1,260 @@
+    [git_commit_behind]
+    user root
+    env.git_path /path/to/git
+
+Then, for each repository you want to check, you need the following
+configuration block under the git_commit_behind section
+    env.repo.[repoCode].path /path/to/local/repo
+    env.repo.[repoCode].name Repo Name
+    env.repo.[repoCode].user user
+    env.repo.[repoCode].warning 10
+    env.repo.[repoCode].critical 100
+
+[repoCode] can only contain letters, numbers and underscores.
+
+path : mandatory, the local path to your git repository
+name : optional (default : [repoCode]), a cleaner name that will be displayed
+user : optional (default : empty), the owner of the repository
+       if set and different from the user running the plugin, the git commands
+       will be executed as this user
+warning : optional (default 10), the warning threshold
+critical : optional (default 100), the critical threshold
+
+For example :
+
+    [git_commit_behind]
+    user root
+
+    env.repo.munin_contrib.path /opt/munin-contrib
+    env.repo.munin_contrib.name Munin Contrib
+
+    env.repo.other_repo.path /path/to/other-repo
+    env.repo.other_repo.name Other Repo
+
+=head1 MAGIC MARKERS
+
+  #%# family=auto
+  #%# capabilities=autoconf
+
+=head1 VERSION
+
+1.0.0
+
+=head1 AUTHOR
+
+Neraud (https://github.com/Neraud)
+
+=head1 LICENSE
+
+GPLv2
+
+=cut"""
+
+
+from __future__ import print_function
+
+import logging
+import os
+from random import randint
+import re
+from subprocess import check_output, call, DEVNULL, CalledProcessError
+import sys
+import time
+
+
+plugin_version = "1.0.0"
+
+debug = int(os.getenv('MUNIN_DEBUG', os.getenv('DEBUG', 0))) > 0
+if debug:
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(asctime)s %(levelname)-7s %(message)s')
+
+conf = {
+    'git_path':  os.getenv('git_path', '/usr/bin/git'),
+    'state_file': os.getenv('MUNIN_STATEFILE',
+                            '/var/lib/munin-node/plugin-state/nobody/' +
+                            'git_commit_behind.state')
+}
+
+repo_codes = set([re.search('repo\.([^.]+)\..*', elem).group(1)
+                  for elem in os.environ.keys() if 'repo.' in elem])
+
+repos_conf = {}
+for code in repo_codes:
+    repos_conf[code] = {
+        'name':        os.getenv('repo.%s.name' % code, code),
+        'path':        os.getenv('repo.%s.path' % code, None),
+        'user':        os.getenv('repo.%s.user' % code, None),
+        'warning':     os.getenv('repo.%s.warning' % code, '10'),
+        'critical':    os.getenv('repo.%s.critical' % code, '100')
+    }
+
+
+def print_config():
+    print('graph_title Git repositories - Commits behind')
+
+    print('graph_args --base 1000 -r --lower-limit 0')
+    print('graph_vlabel number of commits behind')
+    print('graph_scale yes')
+    print('graph_info This graph shows the number of commits behind' +
+          ' for each configured git repository')
+    print('graph_category system')
```

Comment:
I could imagine, that `filetransfer` could be a more suitable category for this plugin.
What do you think?
Otherwise just leave it as it is ...

Reply:
The apt plugins are in `system`, so that's the one I've selected by default.
In my mind, `file_transfer` makes more sense for downloads (wget, torrent, newsgroup), backups, rsync ...

I don't mind changing it, if you think `file_transfer` is better suited, I'll update it.

Reply:
Given that a git repository checkout can be used for a variety of purposes, I would prefer `file_transfer`.